p2p.go

```diff
@@ -574,11 +582,10 @@ func (h *Host) Connect(ctx context.Context, target core.PeerAddrInfo) error {
 
 // Broadcast sends a message to the hosts who subscribe the topic
 func (h *Host) Broadcast(ctx context.Context, topic string, data []byte) error {
-	pub, ok := h.pubs[topic]
-	if !ok {
-		return nil
+	if len(h.pubsub.ListPeers(topic)) == 0 {
+		return ErrNoPeersToBroadcast
```

Comment:
also for fan-out mode, Host did not register the topic it wants to broadcast to, would it fall into here and return error?